src/controllers/accommodation.controller.js

```diff
@@ -0,0 +1,135 @@
+import models from '../models';
+import Response from '../utils/response';
+import AccommodationRepository from '../repositories/accommodation.repository';
+import DbErrorHandler from '../utils/dbErrorHandler';
+import AuthUtils from '../utils/auth.utils';
+
+const { Accommodation } = models;
+
+class Accommodations {
+  static async createAccommodation(req, res) {
+    let response;
+    try {
+      const { userData } = req;
+      const {
+        name,
+        address,
+        description,
+        image,
+        geoLocation,
+        rooms,
+        services,
+        amenities,
+        status
+      } = req.body;
+      const data = {
+        user_id: userData.id,
+        name,
+        address,
+        description,
+        image,
+        geo_location: geoLocation,
+        rooms,
+        services,
+        amenities,
+        status
+      };
+      const { dataValues } = await Accommodation.create(data);
+      if (dataValues) {
+        response = new Response(res, 201, 'Your request to create an accommodation has been sent successfully, wait for approval', dataValues);
```

Comment:
Do we need an approval @shaazk ? if yes from who?

Reply:
Yes, the super admin has to approve the accommodation request from the host-supplier like the description suggests, there will have to be a background check of an accommodation before it is approved so that is what i implemented